src/ValidatorKeysTool.cpp

```diff
@@ -152,6 +155,82 @@ void createRevocation (boost::filesystem::path const& keyFile)
     std::cout << std::endl;
 }
 
+void attestDomain(ripple::ValidatorKeys const& keys)
+{
+    if (keys.domain().empty())
+    {
+        std::cout << "No attestation is necessary if no domain is specified!\n";
+        return;
+    }
+
+    std::cout << "The domain attestation token is:\n";
+
+    // Nothing up our sleeve: this is the amendment identifier of the amendment
+    // which introduced the on-ledger manifest feature on XRP Ledger. It is the
+    // first 32 bytes of the SHA512 of the string "OnLedgerManifests":
+
+    using namespace ripple;
+
+    std::cout << "\n";
+    std::cout << "[[domain-attestation]]\n";
+    std::cout << "pubkey=\"" << toBase58 (TOKEN_NODE_PUBLIC, keys.publicKey()) << "\"\n";
+    std::cout << "domain=\"" << keys.domain() << "\"\n";
+    std::cout << "signature=\"" << keys.sign (
+        "15C4447A07A0755204720D67C6C7C8E4E337682DA3505A57754C9267F6D1256B:" +
+        keys.domain() + ":" +
+        toBase58 (TOKEN_NODE_PUBLIC, keys.publicKey())) << "\"\n\n";
+}
+
+void attestDomain(boost::filesystem::path const& keyFile)
+{
+    using namespace ripple;
+
+    auto keys = ValidatorKeys::make_ValidatorKeys (keyFile);
+
+    if (keys.revoked())
+        throw std::runtime_error (
+            "Operation error: The specified master key has been revoked!");
+
+    attestDomain(keys);
+}
+
+void setDomain (std::string const& domain,
+    boost::filesystem::path const& keyFile)
+{
+    using namespace ripple;
+
+    auto keys = ValidatorKeys::make_ValidatorKeys (keyFile);
+
+    if (keys.revoked())
+        throw std::runtime_error (
+            "Operation error: The specified master key has been revoked!");
+
+    if (domain == keys.domain())
+    {
+        std::cout << "The domain name was already set.\n";
+        return;
+    }
+
+    keys.domain(domain);
+
+    if (domain.empty())
+    {
+        std::cout << "The domain name has been cleared.\n";
+        std::cout << "If you had published a domain attestation token in the ";
+        std::cout << "past, please remove it now.\n";
+    }
+    else
+    {
+        std::cout << "The domain name has been set.\n";
+        std::cout << "You should publish a domain attestation token to allow ";
+        std::cout << "others to associate this validator with your domain.";
+        attestDomain(keys);
+    }
+
+    // Update key file with new token sequence
+    keys.writeToFile (keyFile);
```

Comment:
Suggestion: Write the file before printing the output, so the data won't be lost in the extremely unlikely case that the app crashes during the output.